src/Form/QueueContentsForm.php

```diff
@@ -0,0 +1,173 @@
+      ->setInitMessage(t('Starting Queueing Process.'));
+
+    foreach ($types as $type) {
+      $batch_builder->addOperation([$this, 'batchedQueueItemsForMediaType'], ['media_type_id' => $type]);
+    }
+    batch_set($batch_builder->toArray());
+  }
+
+  /**
+   * Operation callback to queue each mpx media type for imports.
+   *
+   * @param string $media_type_id
+   *   The media type id to process in this callback.
+   * @param array $context
+   *   Context data passed between batches.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeDoesNotExistException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeNotAssociatedWithMpxException
+   */
+  public function batchedQueueItemsForMediaType(string $media_type_id, array &$context) {
+    /* @var \Drupal\media_mpx\Service\QueueVideoImports $service */
+    $service = \Drupal::getContainer()->get('media_mpx.service.queue_video_imports');
+    $batch_size = 200;
+
+    if (empty($context['sandbox'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['last_index'] = 0;
+      $context['results'] = [
+        'success' => 0,
+        'errors' => 0,
+      ];
+    }
+
+    $request = new QueueVideoImportsRequest($media_type_id,
+      $batch_size,
+      $context['sandbox']['last_index'] + 1);
+
+    $response = $service->execute($request);
+
+    if (!isset($context['sandbox']['total_results'])) {
+      $context['sandbox']['total_results'] = $response->getIterator()->getList()->getTotalResults();
+    }
+    $total_items = $context['sandbox']['total_results'];
+
+    $context['results']['success'] += $response->getVideosQueued();
+    $context['results']['errors'] += $response->getErrors();
+    $context['sandbox']['progress'] += $response->getVideosQueued();
+    $context['sandbox']['last_index'] = $context['sandbox']['last_index'] + $batch_size;
+
+    $context['message'] = 'Processed a total of ' . $context['sandbox']['progress'] . ' items.';
+    $context['message'] = t('Queued @progress %type items out of @total.', [
+      '@progress' => $context['sandbox']['progress'],
+      '@total' => $total_items,
+      '%type' => $media_type_id,
+    ]);
+
+    if ($context['sandbox']['progress'] != $total_items) {
+      $context['finished'] = $context['sandbox']['progress'] / $total_items;
+    }
+  }
+
+  /**
+   * Finished callback for the batch queuing operations to update mpx videos.
+   *
+   * @param bool $success
+   *   Result of the queuing operations.
+   * @param array $results
+   *   An array with the count of successful and errored queue operations.
+   */
+  public function batchFinished(bool $success, array $results) {
+    $finished_message = t('Finished with errors.');
+    $messenger = \Drupal::messenger();
+
+    if ($success) {
+      $finished_message = \Drupal::translation()->formatPlural(
```

Comment:
`$this->formatPlural()` here.

Reply:
🤦‍♂ . I thought I couldn't do this because being callbacks, the class doesn't keep any of the actual dependencies passed in construct (or above layers). Rather conveniently this doesn't affect the translation service (nor the `\Drupal::messenger()` above which I just cleaned up as well) because that's provided statically from a Trait, instead of via DI